modules/storages/app/components/storages/admin/storage_view_component.html.erb

```diff
@@ -16,21 +16,25 @@
       component.with_row(scheme: :neutral, color: :muted) do
         grid_layout('op-storage-view--row', tag: :div, align_items: :center) do |grid|
           grid.with_area(:item, tag: :div, mr: 3) do
-            render(Primer::Beta::Text.new(font_weight: :semibold, mr: 1)) { I18n.t('storages.file_storage_view.oauth_applications') }
+            render(Primer::Beta::Text.new(font_weight: :semibold, mr: 1)) { I18n.t('storages.file_storage_view.oauth_configuration') }
           end
         end
       end
 
       component.with_row(scheme: :default) do
-        if storage.new_record? || openproject_oauth_application_section_closed?
-          render(Storages::Admin::OAuthApplicationInfoComponent.new(oauth_application:, storage:))
+        if wizard_step == "oauth_application"
+          render(Storages::Admin::OAuthApplicationInfoCopyComponent.new(oauth_application:, storage:, in_wizard: true))
         else
-          render(Storages::Admin::OAuthApplicationInfoCopyComponent.new(oauth_application:, storage:))
+          render(Storages::Admin::OAuthApplicationInfoComponent.new(oauth_application:, storage:))
         end
       end
 
       component.with_row(scheme: :default) do
-        render(Storages::Admin::OAuthClientInfoComponent.new(oauth_client: storage.oauth_client, storage:))
+        if wizard_step == "oauth_client"
+          render(Storages::Admin::Forms::OAuthClientFormComponent.new(oauth_client: storage.oauth_client, storage:, in_wizard: true))
+        else
+          render(Storages::Admin::OAuthClientInfoComponent.new(oauth_client: storage.oauth_client, storage:))
+        end
```

Comment:
You probably tried, but let me ask anyway: have you tried to move all these conditionals to the component itself (the Object)?

Reply:
I added another commit that drives the refactoring further. While that keeps the conditional inside the `StorageViewComponent`, it reduces the number of conditionals for that purpose to one instead of many.